nginx/config.go

```diff
@@ -158,15 +150,15 @@ http {
 		}
 	}
 
-	{{range $appConfig := $routerConfig.AppConfigs}}{{range $domain := $appConfig.Domains}}server {
-		listen 8080{{ if $routerConfig.UseProxyProtocol }} proxy_protocol{{ end }};
+	{{range $appConfig := $routerConfig.AppConfigs}}{{range $domain := $appConfig.Domains}}{{ range $port := $appConfig.ServicePorts }}server {
+		listen {{ $port }}{{ if $routerConfig.UseProxyProtocol }} proxy_protocol{{ end }};
 		server_name {{ if contains "." $domain }}{{ $domain }}{{ else if ne $routerConfig.PlatformDomain "" }}{{ $domain }}.{{ $routerConfig.PlatformDomain }}{{ else }}~^{{ $domain }}\.(?<domain>.+)${{ end }};
 		server_name_in_redirect off;
 		port_in_redirect off;
 		set $app_name "{{ $appConfig.Name }}";
 
 		{{ if index $appConfig.Certificates $domain }}
```

Comment:
Note to self: need to handle two cases here:

1. that the port is port 80 (the web port)
2. that the port is not port 443 (since we're already asking the app to listen to port 443 and assume the user is handling certs themselves from within the app)